edb/schema/utils.py

```diff
@@ -37,6 +37,7 @@
     from . import schema as s_schema
     from . import types as s_types
     from edb.common import parsing
+    from edb.edgeql.compiler.context import ContextLevel
```

Comment:
I should have thought of this when we talked earlier, but I'd like to try to avoid having schema utils operate on pretty internal edgeql->IR compiler stuff like the context. I think it would be best to do the `context -> ObjectType` resolution in a helper in the edgeql compiler somewhere (`edb.edgeql.compiler.schemactx`, next to the enrich callsite, probably), and just pass in an `ObjectType`

Reply:
Oh, actually, looking a little closer, `enrich_...` already has an argument `collection`, which takes a collection of objects to consider instead of looking it up, and it's used for suggesting pointers when the dot *has* been included. 
Look into whether unifying those cases makes things nicer? (It might not)
(Which could probably be done by passing in a collection in both places, or just passing in the source in both places?)

Reply:
I've refactored the code so instead of `collection`, `pointer_parent: so.Object` is passed as an argument. We still need separate function for finding pointers because they should not be type checked, because missing dot is causing wrong type to be expected.